lib/avromatic/model/types/abstract_timestamp_type.rb

```diff
@@ -14,7 +14,7 @@ def coerce(input)
           elsif input.is_a?(::Time) || input.is_a?(::DateTime)
             coerce_time(input)
           else
-            raise Avromatic::Model::CoercionError.new("Could not coerce '#{input.inspect}' to #{self.class.name.demodulize}")
+            raise ArgumentError.new("Could not coerce '#{input.inspect}' to #{name}")
```

Comment:
Is it necessary to interpolate values and specify an error message here, since the error will always be rescued and replaced by a `CoercionError` that does not use the original error message?